ntbs-service/wwwroot/source/Components/ValidateRequiredCheckboxes.ts

```diff
@@ -0,0 +1,49 @@
+import Vue from 'vue';
+import { getHeaders, getValidationPath } from '../helpers';
+const axios = require('axios');
+const qs = require('qs');
+
+const ValidateRequiredCheckboxes = Vue.extend({
+    props: ['model', 'property', 'shouldvalidatefull'],
+    methods: {
+      validate: function (event: FocusEvent) {
+        // Our onBlur validate events happen on input fields;;
+        const inputField = event.target as HTMLInputElement
+        const newValue = inputField.value;
+
+        var checkboxes = Array.from(this.$refs["checkboxgroup"].getElementsByClassName("nhsuk-checkboxes__input"));
+        var checkboxList = checkboxes
+          .filter((x: any) => x.checked)
+          .map((x : any) => x.id.replace("NotificationSiteMap_", "").slice(0, -1))

```

Comment:
Are you using `slice` here to make sure the string is a new copy? If so, I think `replace` already does that? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace